src/Faker/Provider/fa_IR/Person.php

```diff
@@ -134,4 +134,82 @@ class Person extends \Faker\Provider\Person
 
     protected static $titleMale = array('آقای', 'استاد', 'دکتر', 'مهندس');
     protected static $titleFemale = array('خانم', 'استاد', 'دکتر', 'مهندس');
+
+    /**
+     * this method returns a valid Iranian nationalCode
+     * @return string
+     * @example '8075859741'
+     */
+    public static function nationalCode()
+    {
+        $area = self::createAreaCode();
+        $core = self::createCoreCode();
+        $control = self::createControlCode($area, $core);
+
+        return sprintf("%03d%06d%01d", $area, $core, $control);
+    }
+
+    /**
+     * @return int|string
+     * this method generates a 3-digit valid area code to be used in nationalCode
+     */
+    private static function createAreaCode()
+    {
+        $area = 0;
+
+        while (self::sumOfDigits($area) == 0) {
+            $area = mt_rand(0, 9) . mt_rand(0, 9) . mt_rand(0, 9);
+        }
+
+        return $area;
+    }
+
+    /**
+     * @param $number
+     * @return int
+     * this helper method sums up the digits in a number
+     */
+    private static function sumOfDigits($number)
+    {
+        $sumOfDigits = 0;
+
+        for ($i = 0; $i < strlen($number); $i++) {
+            $sumOfDigits += $number[$i];
+        }
+
+        return $sumOfDigits;
+    }
+
+    /**
+     * @return string
+     * this method randomly generates a 6-digit core code for nationalCode
+     */
+    private static function createCoreCode()
+    {
+        return mt_rand(1, 9) . mt_rand(1, 9) . mt_rand(1, 9) . mt_rand(1, 9) . mt_rand(1, 9) . mt_rand(1, 9);
+    }
+
+    /**
+     * @param $area
+     * @param $core
+     * @return int
+     * this method uses the Iranian nationalCode validation algorithm to generate a valid 10-digit code
```

Comment:
please link to the specification algorithm. Also, please put the description as the first line in the phpDoc, not the last one

Reply:
Did so.